src/ParallelAlgorithms/Amr/Actions/RegisterCallbacks.hpp

```diff
@@ -0,0 +1,38 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <deque>
+#include <vector>
+
+#include "Parallel/Callback.hpp"
+#include "ParallelAlgorithms/Amr/Actions/CollectDataFromChildren.hpp"
+#include "ParallelAlgorithms/Amr/Actions/Component.hpp"
+#include "ParallelAlgorithms/Amr/Actions/CreateChild.hpp"
+#include "ParallelAlgorithms/Amr/Actions/SendDataToChildren.hpp"
+#include "Utilities/Serialization/RegisterDerivedClassesWithCharm.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace amr {
+template <typename Component>
+void register_callbacks() {
+  using ArrayIndex = typename Component::array_index;
+  register_classes_with_charm(
+      tmpl::list<
+          Parallel::SimpleActionCallback<
+              amr::Actions::CreateChild,
+              CProxy_AlgorithmSingleton<amr::Component<metavariables>, int>,
```

Comment:
`metavariables` isn't defined here, not sure why this doesn't error.

Reply:
It doesn't error because of https://github.com/sxs-collaboration/spectre/blob/144c684133a565a0f28db40edcf8bedf3ec1ea6d/cmake/AddSpectreExecutable.cmake#L85

I've wondered off and on whether we should have given that a less generic name so this sort of thing didn't happen.